### PR TITLE
Enable Windows support

### LINKED ISSUE #88: Windows: 'pthread.h': No such file or directory
Since the native Windows threading API is exposed via `#include <windows.h>`, we need to make some minor changes to support the EDDL on Windows. We should modify: `CMakeLists.txt`, `eddlConfig.cmake.in` and some functions inside the folder `net/` (...although I might be missing something else, check it)

I'll leave this issue for the Windows fans 😉

To build it type this from the source directory:
```
mkdir build; cd build
cmake -G "NMake Makefiles" ..
nmake
nmake install
```

### PR DESCRIPTION
**Pull request by MicheleCancilla**
This PR enables EDDL to run on Windows. 
The modifications can be summarized as:
1. A Windows GitHub workflow has been added, which builds correctly, but one test fails because of #163.
2. The CMakeLists have been modified to support both Linux and Windows.
3. SUPERBUILD has been ported too. Both Release and Debug dependency libraries will be installed by CMake for Windows. I have also added Pthreads for Windows (resolves #88).

Let me know if you agree with the changes made.

### CODE REVIEW

File: .github/workflows/build-cpu-windows.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      matrix:
+        #os: [windows-latest, windows-2016]
+        os: [windows-latest]
+    steps:
+    - uses: actions/checkout@v2
+    - name: Configure
+      run: |
+        mkdir build
+        cd build
+        cmake -DBUILD_SUPERBUILD=ON -DBUILD_SHARED_LIBS=OFF ..
```
Comment (salvacarrion) [inline .github/workflows/build-cpu-windows.yml:18]: Why "-D DBUILD_SHARED_LIBS=OFF"? By default it is ON (master)

**Comment by RParedesPalacios** [comment]
Please, i think that you have to schedule a char with Salva ...

**Comment by salvacarrion** [comment]
Can you change the target branch to "windows"? I'd rather not merge it to master directly.

File: cmake/pthreads4w.CMakeLists.txt.in [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    SET(VCVARS "@CMAKE_GENERATOR_INSTANCE@/VC/Auxiliary/Build/vcvarsamd64_x86.bat")
+endif()
+
+include(ExternalProject)
+ExternalProject_Add(pthreads4w
+    PREFIX pthreads4w
+    URL https://sourceforge.net/projects/pthreads4w/files/pthreads4w-code-v3.0.0.zip/download
+    URL_HASH MD5=f3bf81bb395840b3446197bcf4ecd653
+    SOURCE_DIR "${EP_BASE_DIR}/pthreads4w-src"
+    BUILD_IN_SOURCE 1
```
Comment (salvacarrion) [inline cmake/pthreads4w.CMakeLists.txt.in:22]: Why not use "BINARY_DIR "${EP_BASE_DIR}/pthreads4w-build"

File: cmake/pthreads4w.CMakeLists.txt.in [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    CONFIGURE_COMMAND ""
+    BUILD_COMMAND ${VCVARS} && nmake all install
+    INSTALL_COMMAND ""
```
Comment (salvacarrion) [inline cmake/pthreads4w.CMakeLists.txt.in:25]: If is it possible, I would like to follow the same format in all modules.
Ps.: I don't know how this works. Just asking.

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -79,30 +83,41 @@ target_include_directories(${PROJECT_NAME} PRIVATE ${EIGEN3_INCLUDE_DIRS})
 target_link_libraries(${PROJECT_NAME} PUBLIC Eigen3::Eigen)  # Header only library
 
 # ZLIB
-if(DEFINED ZLIB_INCLUDE_DIRS AND DEFINED ZLIB_LIBRARIES_DIRS AND DEFINED ZLIB_LIBRARIES)
-    message(STATUS "Using specific Zlib")
+if(BUILD_SUPERBUILD AND NOT USE_SYSTEM_ZLIB)
```
Comment (salvacarrion) [inline src/CMakeLists.txt:86]: Sometimes CMake says that it has found ZLIB/GTest/etc, but then some of these variables it's empty

File: .github/workflows/build-cpu-windows.yml (hunk 1 above)
Comment (CostantinoGrana) [reply .github/workflows/build-cpu-windows.yml:18]: To avoid having DLL files in the system path. This is just under Windows.

**Comment by CostantinoGrana** [comment]
The code which allows to compile under Windows should be in the master branch. It also fixes problems in Linux installation. For example `EDDLConfig.cmake.in` didn't export the installation path correctly, so other libraries importing EDDL with CMake cannot find the libraries installed with SUPERBUILD.

File: cmake/pthreads4w.CMakeLists.txt.in (hunk 3 above)
Comment (CostantinoGrana) [reply cmake/pthreads4w.CMakeLists.txt.in:25]: This is not installed with CMake, so it's mandatory to use `nmake`. 

By the way, are you still using pthreads? If not, I'd be happy to remove this dependency.

**Comment by RParedesPalacios** [comment]
Moving to Master branch and cross-finger. I hope that this doesn't break anything for Linux/Mac and only affect to windows installation.

**Pull request merged**